packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -1141,7 +1147,25 @@ contract VotingV2 is
                 voteInstance.resultComputation.getResolvedPrice(_computeGat(priceRequest.lastVotingRound));
             return (true, resolvedPrice, "");
         } else if (requestStatus == RequestStatus.Future) return (false, 0, "Price is still to be voted on");
-        else return (false, 0, "Price was never requested");
+        else {
+            (bool previouslyResolved, int256 previousPrice) =
+                _getPriceFromPreviousVotingContract(identifier, time, ancillaryData);
+            if (previouslyResolved) return (true, previousPrice, "Returned from previous contract");
+            else (false, 0, "Price was never requested");
+        }
+    }
+
+    function _getPriceFromPreviousVotingContract(
+        bytes32 identifier,
+        uint256 time,
+        bytes memory ancillaryData
+    ) public view returns (bool, int256) {
+        if (address(previousVotingContract) == address(0)) return (false, 0);
+        try previousVotingContract.getPrice(identifier, time, ancillaryData) returns (int256 price) {
```

Comment:
nit: why not use `if (previousVotingContract.hasPrice())` rather than try-catch just to avoid any potential gas issues that come with try-catch.

Reply:
done.